libnd4j/include/pointercast.h

```diff
@@ -9,6 +9,7 @@
 
 typedef void* Nd4jPointer;
 typedef long long Nd4jLong;
+typedef long Nd4jInt;
```

Comment:
No way.

Reply:
I'm fine with removing this typedef, i was simply trying to stay as close as possible to the existing especially how LongBuffers are handles.

Reply:
Nd4jLong is declared in this way due to legacy reasons. And FlatBuffers. Eventually it should just become int64_t

Reply:
So should I just use `int32_t`?

Reply:
just int for now

Reply:
Why do we need 32-bit here?

Reply:
Why do we need concatInt in the first place?

Reply:
@saudet and @raver119 , I need to know whether or not I am correct in assuming that using `int` will not work.

`int` in cpp is 16 bit AFAIK, whereas IntBuffer reports elementSize is 4 bytes => 32 bit.
Therefore I must assume I shoud be using either `int_32` or `long`, correct?

Reply:
Java SE doesn't work on platforms with less than 32 bits and `int` is pretty much guaranteed to be 32-bit long on those these days, so `int` is fine. Anyway, we're using a `typedef` in case we want to change it, so not a big issue.

Reply:
> Anyway, we're using a typedef in case we want to change it, so not a big issue.

is int a typedef?  or are you referring to the `Nd4jInt` typedef that @raver119  told me to remove?

Reply:
No, he asked you to change it to `typedef int Nd4jInt`.

Reply:
I'm sorry to drag this out, but in in an earlier [comment](https://github.com/deeplearning4j/deeplearning4j/pull/5686#discussion_r197520132) he said:
> We don't need Nd4jInt definition.

So I'm still a bit confused :/

Reply:
It's not a big deal, let's start by doing something that works! "long" is 64-bit on non-Windows system so that's not going to work for sure.

Reply:
Gotcha! :)
I will replace with  `typedef int Nd4jInt` and remove the cuda attr stuff.